client/src/components/CKANResourceLocatorField/URLInput.js

```diff
@@ -38,7 +38,6 @@ class URLInput extends Component {
     const props = {
       className: classNames(
         className,
-        'no-change-track',
```

Comment:
I'm not convinced that this is the right solution. The ideal solution is that once the field has a valid value then the form is marked as changed. This _sort of_ works already because the changetracker does pick up changes in the hidden field (which is only populated when it has a valid value). The problem is that we need to give entwine a kick for the changetracker to actually notice. That's why you can make it see changes when clicking on a field or something.

We had this issue in elemental too so I think there's probably some code we can look at.

Reply:
Ok - I'm in two minds about that. I think you should be able to save an invalid value, but it would keep displaying the validation error on the field if you do so. Either way it should match core behaviour for form fields

Reply:
Hmmmn. At the moment when you save an invalid "locator field" it will not persist whatever you've put in the URLInput field. Maybe we should change that then? 

From a semantics POV I agree with allowing to store an invalid value when you're talking about setting a prop on a model in code, but I think it's kind of weird to be able to save a form with errors on it on the client side.

Reply:
Perhaps the problem is that changing the text field doesn't immediately update the hidden input value, which is what _should_ be change tracked

Reply:
I haven't looked into it too much but I _think_ it'll be because entwine is not picking up the change one the hidden input made by React. That's why you get the save button activating when you open the dropdown (but not choosing a value yet).

I'm pretty keen to look at this and potentially come up with a generic solution that can go together with our idea of a generic "JSON" field type (raised in #65)

Reply:
Ok I prepped a patch that you might not like (😂) that fixes this issue:
```diff
--- client/src/components/CKANResourceLocatorField.js	(revision 3787e7d7da82db8795a2587965b9153725b5e1e4)
+++ client/src/components/CKANResourceLocatorField.js	(date 1547516902000)
@@ -31,6 +31,8 @@
       forceInvalidTimeout: null,
     };
 
+    this.valueInput = null;
+
     this.handleChange = this.handleChange.bind(this);
     this.handleResourceSelect = this.handleResourceSelect.bind(this);
 
@@ -47,6 +49,14 @@
     }
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (this.valueInput && JSON.stringify(prevState.spec) !== JSON.stringify(this.state.spec)) {
+      const event = new Event('change', { bubbles: true });
+      event.simulated = true;
+      this.valueInput.dispatchEvent(event);
+    }
+  }
+
   /**
    * Indicates the URL inpit field is invalid and provides a localised message indicating the issue
    * Currently this shows a single message in cases where the field is invalid but it could be
@@ -291,7 +301,14 @@
     const { spec, validationMessage } = this.state;
     const value = !spec || validationMessage ? null : JSON.stringify(spec);
 
-    return <Input name={this.props.name} type="hidden" value={value} />;
+    return (
+      <Input
+        name={this.props.name}
+        type="hidden"
+        value={value}
+        innerRef={input => { this.valueInput = input; }}
+      />
+    );
   }
 
   /**
```

Missing some comments but React doesn't trigger change events on inputs when values change for some "dedupe"-ing reasons so the idea is to trigger it yourself. See this comment: https://github.com/cypress-io/cypress/issues/536#issuecomment-308739206 that I dug up with some Googling.

Reply:
Go for it

Reply:
Cool. I've updated this PR with the patch and rebased it.